pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -84,6 +84,9 @@ buildPythonPackage rec {
 
   __darwinAllowLocalNetworking = true;
 
+  # fix on current master
+  patches = [ ./fix_test_generic_options.patch ];
```

Comment:
> I had to edit the patch to get from starting from pyarrow instead of python/pyarrow.

I suppose this can still be done with a proper combination of `stripLen` and/or `extraPrefix`.

Reply:
not sure what difference would it make.

Reply:
I think you can use [fetchpatch](https://nixos.org/manual/nixpkgs/unstable/#fetchpatch) with `stripLen` to make the upstream patch to apply.

Reply:
fixed.